shared/src/enums/app-environment.enum.ts

```diff
@@ -1,6 +1,8 @@
-enum AppEnvironment {
-    DEVELOPMENT = 'development',
-    PRODUCTION = 'production',
-}
+const AppEnvironment = {
+    DEVELOPMENT: 'development',
+    PRODUCTION: 'production',
+} as const;
+
+type AppEnvironment = typeof AppEnvironment;
```

Comment:
Throughout the entire pull request, whenever you need to specify that value has enum type use ValueOf<typeof YOUR_ENUM_HERE>.